src/LEDUtils.cpp

```diff
@@ -1,13 +1,34 @@
 #include "LEDUtils.h"
 
 cRGB
-breath_compute(uint8_t hue, uint8_t saturation) {
+breath_compute(uint8_t hue, uint8_t saturation, uint8_t phaseOffset) {
+  // The phase offset is provided in case one wants more than one breathe effect
+  // differing in phase at the same time. This may be useful for individual
+  // indicators that need to contrast with any other overall breathe effect.
+  // (Offset value 128 giving 50% ie opposite phase is especially nice.)
+  //
+  // The actual breathe computation function has a period of 4096. The input
+  // phase offset ranges from 0 to 255. This is to be scaled to the actual
+  // period 0 to 4096. To allow precise offsets of 25%, 50% etc, and for
+  // computational convenience and speed (and since one would not meaningfully
+  // request an offset of 4096 for 255 to be mapped to it), we merely multiply
+  // the input offset by 16 (== 4096/256), or equivalently lshift it by 4.
+
+  return breath_compute_helper(hue, saturation, millis() + (phaseOffset << 4));
+  // The actual computation is in a separate helper to provide for multiple
+  // breathe calculations with different hue/saturation input but the same time
+  // for the effect to be synced and avoiding multiple calls to millis()
```

Comment:
Wouldn't it make the code slightly easier if `millis()` would be replaced by `Kaleidoscope.millisAtCycleStart()`? The latter is guaranteed to remain the same for the duration of the cycle, and as such, should nicely keep multiple breath effects in sync.

As far as I see, if we did this, there wouldn't be a need for `breath_compute_helper()`, unless one would want to have breathe effects that are deliberately out of sync. I can imagine use-cases for that too, but... the solution for that is to call `breathe_compute` at different times. In other words, doing the out-of-sync delay on the caller side.

Does this make sense?